packages/askui-nodejs/src/core/reporting/step-reporter.ts

```diff
@@ -4,68 +4,121 @@ import { Reporter } from './reporter';
 import { ReporterConfig } from './reporter-config';
 import { DefaultStep } from './default-step';
 import { logger } from '../../lib/logger';
+import { maxSnapshotDetailLevel } from './snapshot-detail-level';
+
+interface ReporterWithPromiseQueue {
+  reporter: Reporter;
+  promiseQueue: Promise<void>;
+}
+
+function buildReportersWithPromiseQueue(
+  reporter: Reporter | Reporter[] | undefined,
+): ReporterWithPromiseQueue[] {
+  if (reporter === undefined) {
+    return [];
+  }
+  if (Array.isArray(reporter)) {
+    return reporter.map((r) => ({
+      reporter: r,
+      promiseQueue: Promise.resolve(),
+    }));
+  }
+  return [
+    {
+      reporter,
+      promiseQueue: Promise.resolve(),
+    },
+  ];
+}
 
 export class StepReporter {
-  currentStep?: DefaultStep | undefined;
+  private _currentStep?: DefaultStep | undefined;
+
+  private reporters: ReporterWithPromiseQueue[];
+
+  readonly config: Required<ReporterConfig>;
 
-  constructor(private reporter: Required<Reporter> & { config: Required<ReporterConfig> }) {}
+  constructor(reporter?: Reporter | Reporter[] | undefined) {
+    this.reporters = buildReportersWithPromiseQueue(reporter);
+    this.config = {
+      withScreenshots: maxSnapshotDetailLevel(
+        ...this.reporters.map(
+          (r) => r.reporter.config?.withScreenshots ?? 'onFailure',
+        ),
+      ),
+      withDetectedElements: maxSnapshotDetailLevel(
+        ...this.reporters.map(
+          (r) => r.reporter.config?.withDetectedElements ?? 'onFailure',
+        ),
+      ),
+    };
+  }
+
+  flush(): Promise<void> {
+    return Promise.all(
+      this.reporters.map((reporter) => reporter.promiseQueue),
+    ).then(() => {});
+  }
 
-  get config(): Required<ReporterConfig> {
-    return this.reporter.config;
+  get currentStep(): DefaultStep | undefined {
+    return this._currentStep;
   }
```

Comment:
is this being used ?

Reply:
Yes, in the test of the corresponding class.